R/voter.R

```diff
@@ -0,0 +1,155 @@
+# voter.R
+#
+# Implementation of voter model dynamics on a network in R.
+# author: Luke Perry
+# Submitted as part of research assistance for Dr. William Fagan and Anshuman Swain
+#
+# Simulate voter-model-style dynamics on a network.
+# Nodes are randomly assigned a state in :math:`\{-1, 1\}`; at each
+# time step all nodes asynchronously update by choosing their new
+# state uniformly from their neighbors. Generates an :math:`N \times
+# L` time series.
+# The results dictionary also stores the ground truth network as
+# `'ground_truth'`.
+#
+# Parameters
+# ----------
+# G (matrix)
+#     the input (ground-truth) graph with `N` nodes.
+# L (int)
+#     the length of the desired time series.
+# noise (float, str or None)
+#     if noise is present, with this probability a node's state will
+#     be randomly redrawn from :math:`\{-1, 1\}` independent of its
+#     neighbors' states. If 'automatic', set noise to :math:`1/N`.
+#
+# Returns
+# -------
+# TS (matrix)
+#     an :math:`N \times L` array of synthetic time series data.
+
+
+# dependency import
+library(igraph)
+
+# get dimensions helper function
+DIM <- function( ... ){
+  args <- list(...)
+  lapply( args , function(x) { if( is.null( dim(x) ) )
+    return( length(x) )
+    dim(x) } )
+}
+
+voter <- function(G, L, noise=NULL) {
+  # get adj matrix
+  G = graph_from_adjacency_matrix(x, weighted = TRUE)

```

Comment:
When using an exported function there are a couple of ways we need to import it.  You can do it using [comments](https://kbroman.org/pkg_primer/pages/depends.htmlhttps://kbroman.org/pkg_primer/pages/depends.html).  You can also just explicitly reference the namespace like `igraph::graph_from_adjacency_matrix` which is what I usually do.